app/talk/board.cjsx

```diff
@@ -61,8 +65,44 @@ module.exports = React.createClass
   setDiscussions: (page = @props.location.query.page) ->
     @discussionsRequest(page)
       .then (discussions) =>
+        subject_ids = []
+        author_ids = []
         discussionsMeta = discussions[0]?.getMeta()
         @setState {discussions, discussionsMeta, loading: false}
+        discussions.map (discussion) ->
+          subject_ids.push discussion.focus_id if discussion.focus_id and discussion.focus_type is 'Subject'
+          author_ids.push discussion.latest_comment.user_id
+
+        apiClient
+          .type 'users'
+          .get
+            id: author_ids
+          .then (users) =>
+            users.map (user) => 
+              @setState (prevState, props) ->
```

Comment:
This is pretty cool. Seems like an efficient way to update the state of a list of items.